File: Supero.TaskList.Web/src/app/models/TaskList.ts
export class TaskList {
    constructor(public Id?: number,
        public IdUsuario?: number,
        public Titulo?: string,
        public Descricao?: string,
        public StatusTaskList?: number,
        public DataCriacao?: Date ,
        public DataAlteracao?: Date  ,
        public DataExclusao?: Date  ,
        public DataConclusao?: Date,
        public Itens?: TaskListItem[],
        public Status?: StatusTaskListEnum) { }
}

export class TaskListItem {
    constructor(public Id?: number,
        public Descricao?: string,
        public IdTaskList?: number,
    ) { }
}

export enum StatusTaskListEnum {
    Ativa = 0,
    Pendente = 1,
    Cancelada = 2,
    Concluida = 3
}

File: Supero.TaskList.Data/Models/TaskList.cs
using System;
using System.Collections.Generic;

namespace Supero.TaskList.Data.Models
{
    public partial class TaskList : IEntity
    {
        public TaskList()
        {
            this.TaskListItems = new List<TaskListItem>();
        }

        public int Id { get; set; }
        public string IdUsuario { get; set; }
        public string Titulo { get; set; }
        public string Descricao { get; set; }
        public byte IdStatus { get; set; }
        public System.DateTime DataCriacao { get; set; }
        public Nullable<System.DateTime> DataAlteracao { get; set; }
        public Nullable<System.DateTime> DataExclusao { get; set; }
        public Nullable<System.DateTime> DataConclusao { get; set; }
        public virtual ICollection<TaskListItem> TaskListItems { get; set; }
    }
}
